lib/scholar/cluster/gmm.ex

```diff
@@ -0,0 +1,407 @@
+defmodule Scholar.Cluster.GaussianMixture do
+  @moduledoc """
+  Gaussian Mixture Model.
+
+  Gaussian Mixture model is a probabilistic model that assumes every data point is generated
+  by randomly choosing one of the several fixed Gaussian distributions and then sampling from it.
```

Comment:
"randomly choosing" might be a loaded expression here. I got stuck on thinking about if this is random uniform, normal or whatever else method of choosing.

```suggestion
  b choosing one of several fixed Gaussian distributions and then sampling from it.
```

Reply:
Yes, it is a multinomial random distribution. Basically GMM can be viewed as a generative model in the following way:
- First sample one out of K Gaussian distributions from multinomial random distribution with parameters w_1, ..., w_K (w_k is the probability of sampling k-th Gaussian; this is the `weights` field of the model).
- Then sample from that Gaussian distribution.

Reply:
Agreed about the suggestion though.